sequencer.c

```diff
@@ -2331,11 +2331,10 @@ static int do_pick_commit(struct repository *r,
 		*check_todo = !!(flags & EDIT_MSG);
 		if (!res && reword) {
 fast_forward_edit:
-			if (item->arg_len > 0 && (strlen(msg.subject) !=
-			    item->arg_len ||
-			    strncmp(msg.subject, todo_item_get_arg(todo_list,
-				    item),
-				    item->arg_len) != 0)) {
+			if (opts->allow_inline_reword && item->arg_len > 0 &&
+			    (strlen(msg.subject) != item->arg_len ||
+			     strncmp(msg.subject, todo_item_get_arg(todo_list,
+				     item),item->arg_len) != 0)) {
```

Comment:
It would probably make most sense to move the new `allow_inline_reword` attribute as well as this hunk to the first commit, so that the patch series does not read like "let's introduce that feature! Oh never mind, let's make it opt-in!"